src/script/campaign/mod/vco-listeners.lua

```diff
@@ -155,6 +161,23 @@ end
 local function add_listeners()
 	vco:log("Adding listeners");
 
+	vco:log("- Moulder listeners");
+    core:add_listener(
+        "vco_skv_moulder_upgrade_count_check",
+        "UnitTrained",
+        function(context)
+            vco:log("inside mdr listener conditional check")
+            local faction = context:unit():faction()
+            return faction:name() == "wh2_main_skv_clan_moulder" and faction:is_human()
+        end,
+        function(context)
+            vco:log("inside mdr listener execution")
```

Comment:
Not checking the code/logs yet because I assume it is WIP, but will come back once the release is ready. In any case, first function is a filter and second unit is the the actual callback. We do not want callbacks to hold any code, so just leave a function there. If it has any params, use something like:
```lua
function(context)
	check_vco_brt_bordeleaux_alberic_vow(context:character());
end,
```
else
```lua
check_vco_brt_bordeleaux_alberic_vow,
```